src/pages/index.js

```diff
@@ -97,15 +156,41 @@ function renderCard(cardData) {
 function handleProfileFormSubmit(inputValues) {
   // utilize the inputValues object
   // and call the setUserInfo method (in the UserInfo class) to set the name and description that was typed in by the user
-  userInfo.setUserInfo(inputValues.name, inputValues.description);
+  return api
+    .editProfile({ name: inputValues.name, about: inputValues.description })
+    .then((res) => {
+      userInfo.setUserInfo(res.name, res.about);
+    })
+    .catch(console.error);
 }
 
 function handleAddCardFormSubmit(inputValues) {
   const name = inputValues.title;
   const link = inputValues.url;
-  renderCard({ name, link });
-  cardFormElement.reset();
-  cardFormValidator.toggleSubmitButton();
+  if (!link.startsWith("http://") && !link.startsWith("https://")) {
+    console.error("Please enter a valid URL starting with http:// or https://");
+    return;
+  }
+  api
```

Comment:
* When I add a new card, the button text doesn't change to `Saving...`


You should add **return** here, to continue the `promise` inside the `PopupWithForm` class

```js
// return  lets us use `.then` and `.finally` after that
  return api
    .addCard({
      name: name,
```